### ISSUE TITLE
PrintStyledContent with attribute resetting background color

### ISSUE THREAD

**Issue by carlocaione** [crossterm-rs/crossterm#424]
`crossterm = "0.17"`

Consider the following example code:
```
let styled_foo = style::style("foo");
let styled_bar = style::style("bar").attribute(Attribute::Bold);

queue!(
    stdout,
    style::SetBackgroundColor(Color::Black),
    style::PrintStyledContent(styled_foo),
    style::PrintStyledContent(styled_bar),
)?;

stdout.flush()?;
```
The output is the one expected:
![Screenshot from 2020-05-01 19-17-34](https://user-images.githubusercontent.com/357758/80825332-729c4300-8be0-11ea-9265-a4abf3f7dca7.png)

Now we modify the code changing only:
```
let styled_foo = style::style("foo").attribute(Attribute::Bold);
let styled_bar = style::style("bar").attribute(Attribute::Bold);
```
The output is now (wrong):
![Screenshot from 2020-05-01 19-18-56](https://user-images.githubusercontent.com/357758/80825419-a37c7800-8be0-11ea-8f08-9fb9fa46b923.png)
The green background is lost.

Is this expected?

**Comment by TimonPost** [comment]
It resets all style, yes this is known, however, it came up once before. I would love to see this improved. 

The styling code exists over here:

https://github.com/crossterm-rs/crossterm/blob/master/src/style/styled_content.rs#L94


The reset happens here:

https://github.com/crossterm-rs/crossterm/blob/master/src/style/styled_content.rs#L118

**Comment by carlocaione** [comment]
Right, I see the problem now. I would probably suggest to always reset the colors / attributes then otherwise this could be confusing.

**Comment by TimonPost** [comment]
It is fixed in #483
